lib/archivers/zip/zip-archive-entry.js

```diff
@@ -199,14 +199,10 @@ ZipArchiveEntry.prototype.setTime = function(time) {
 };
 
 ZipArchiveEntry.prototype.setUnixMode = function(mode) {
-  mode &= ~constants.S_IFDIR;
-  var extattr = 0;
-
-  if (!this.isDirectory()) {
-    mode |= constants.S_IFREG;
-  }
+  mode &= ~constants.S_IFMT;
+  mode |= this.isDirectory() ? constants.S_IFDIR : constants.S_IFREG;
 
-  extattr &= ~this.getExternalAttributes();
```

Comment:
@obayer using the previous attributes as the base is best just incase someone sets something custom besides mode.